### PR TITLE
Update: Use Ember concurrency for data interfaces/adapters

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Update facts/dimension adapters and services to use ember-concurrency tasks. Cancels elide queries we no longer use

## Proposed Changes
- Convert most data interfaces/adapters to be task based
- dimension select local search if all results are fetched
- update typescript/ember-concurrency

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/services/navi-facts.ts [hunk 1]
```diff
@@ -66,34 +65,31 @@ export default class NaviFactsService extends Service {
...
+   * @param request - request object
+   * @param options - options object
+   * @returns - url for the request
    */
-  getDownloadURL(request: RequestV2, options: RequestOptions) {
+  @task *getDownloadURL(request: RequestV2, options: RequestOptions): TaskGenerator<string> {
     const { type: dataSourceType } = getDataSource(request.dataSource);
     const adapter = this._adapterFor(dataSourceType);
-    return adapter.urlForDownloadQuery(request, options);
+    return yield adapter.urlForDownloadQuery(request, options);
```
Comment (jkusa) [inline packages/data/addon/services/navi-facts.ts:76]: do we need to call `perform` here

**Review by jkusa — COMMENTED** [review]
Do we have any tests that exercise the cancel functionality?

File: packages/data/addon/adapters/facts/elide.ts [hunk 2]
```diff
@@ -266,40 +268,50 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
+    try {
+      while (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+        yield timeout(this._pollingInterval);
+        asyncQueryPayload = yield this.fetchAsyncQuery(id, request.dataSource);
+        status = asyncQueryPayload?.asyncQuery.edges[0]?.node.status;
+      }
+      return asyncQueryPayload;
+    } finally {
+      if (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+        console.error(`cancelled query ${id}`, asyncQueryPayload);
```
[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/data/addon/adapters/facts/elide.ts]: Removed for now, it was creating a lot of messages

File: packages/data/addon/services/navi-facts.ts (hunk 1 above)
Comment (kevinhinterlong) [reply packages/data/addon/services/navi-facts.ts]: Good catch 👍 , unfortunately the typescript for calling is not very helpful

**Review by jkusa — APPROVED** [review]

**Pull request merged**